assets/test/detail.spec.js

```diff
@@ -0,0 +1,829 @@
+            var statusResult = config.errors.en.missingInformation + '(' + config.errors.en.type.application + ')';
+            var messageResult = config.errors.en.noArticleId;
+            var alertStatus = alertBox.querySelector('.status');
+            var alertMessage = alertBox.querySelector('.message');
+            expect(alertStatus.innerHTML).to.eql(statusResult);
+            expect(alertMessage.innerHTML).to.eql(messageResult);
+        });
+    });
+
+    describe('getArticleSuccess()', function () {
+        var getWindowPathnameStub;
+        var getDetailActionsStub;
+        var renderArticleStub;
+        before(function () {
+            getWindowPathnameStub = sinon.stub(detail, "getWindowPathname", function () {
+                return '/article/00353/1/1345f644-67ba-479d-8e54-65e7803c79c1';
+            });
+            getDetailActionsStub = sinon.stub(detail, "getDetailActions", function () {
+            });
+            renderArticleStub = sinon.stub(detail, "renderArticle", function () {
+            });
+        });
+        after(function () {
+            getWindowPathnameStub.restore();
+            getDetailActionsStub.restore();
+            renderArticleStub.restore();
+        });
+        it('Should update details', function () {
+            detail.setArticleParams();
+            detail.getArticleSuccess(data);
+            expect(detail.detail.article).to.eql(data);
+            expect(detail.detail.currentArticle).to.eql(data.versions[1].details);
+            expect(detail.detail.currentEvents).to.eql(data.versions[1].runs[1]);
+        });
+    });
+
+    describe('getArticleError()', function () {
+        after(function () {
+            $('#article').empty();
+            $('#article-console').empty();
+        });
+        it('Should display standard error alert and error debug alert', function () {
+            var data = {
+                statusCode: 400,
+                statusText: "Error Occurred",
+                responseText: {
+                    "message": "I'm afraid I can't do that Dave",
+                    "detail": "I'm afraid I can't do that Dave",
+                }
+            };
+            detail.getArticleError(data);
+            var alertBox = document.querySelector('#error-message.getArticleError');
+            var statusResult = 'Error Occurred(API Error)';
+            var messageResult = 'I\'m afraid I can\'t do that Dave';
+            var alertStatus = alertBox.querySelector('.status');
+            var alertMessage = alertBox.querySelector('.message');
+            expect(alertStatus.innerHTML).to.eql(statusResult);
+            expect(alertMessage.innerHTML).to.eql(messageResult);
+        });
+    });
+
+    describe('setLatestArticle', function () {
+        after(function () {
+            detail.resetParams();
+        });
+        // var queryParams = {"articleId": "00353", "versionNumber": "1", "runId": "1345f644-67ba-479d-8e54-65e7803c79c1"};
```

Comment:
Please either delete this comment (or if it's really necessary to retain it, add a comment explaining why it's there).